spectator-reg-atlas/src/main/java/com/netflix/spectator/atlas/AtlasMaxGauge.java

```diff
@@ -43,18 +43,19 @@
     this.stat = id.withTag(Statistic.max).withTags(id.tags()).withTag(DsType.gauge);
   }
 
-  @Override public Iterable<Measurement> measure() {
+  @Override void measure(List<Measurement> ms) {
     // poll needs to be called before accessing the timestamp to ensure
     // the counters have been rotated if there was no activity in the
     // current interval.
     double v = value.poll();
-    final Measurement m = new Measurement(stat, value.timestamp(), v);
-    return Collections.singletonList(m);
+    ms.add(new Measurement(stat, value.timestamp(), v));
   }
 
   @Override public void set(double v) {
-    value.getCurrent().max(v);
-    updateLastModTime();
+    if (Double.isFinite(v)) {
```

Comment:
Why does max gauge do the check on set and a regular gauge on measure?

Reply:
Max is activity based and automatically resets to 0.0. This is preventing a single bad recorded value from corrupting the max. Though double checking, this can be removed because the max function already has a sanity check. It is similar to the checks to exclude negative values on timers.

For a regular gauge, it will just report the last value until the TTL. Setting explicitly to `NaN` is common for clearing without waiting so in the measure we just double check and do not emit if it would get dropped later anyway.